src/test/app/Loan_test.cpp

```diff
@@ -3589,6 +3589,26 @@ class Loan_test : public beast::unit_test::suite
                     fee(env.current()->fees().base * 5));
             },
             CaseArgs{.requireAuth = true, .authorizeBorrower = true});
+
+        testCase(
+            [&, this](Env& env, BrokerInfo const& broker, auto&) {
+                using namespace loan;
+                Number const principalRequest = broker.asset(1'000).value();
+                Vault vault{env};
+                auto tx = vault.set({.owner = lender, .id = broker.vaultID});
+                tx[sfAssetsMaximum] = BrokerParameters::defaults().vaultDeposit;
+                env(tx);
+                env.close();
+
+                testcase("Vault maximum value exceeded");
+                env(set(issuer, broker.brokerID, principalRequest),
+                    counterparty(lender),
+                    interestRate(TenthBips32(10'000)),
+                    sig(sfCounterpartySignature, lender),
+                    fee(env.current()->fees().base * 5),
+                    ter(tecLIMIT_EXCEEDED));
+            },
+            nullptr);
```

Comment:
I think it would be good to have another test case where the loan has interest, and the principal request is _under_ the limit, but the interest drives it _over_.

Reply:
This testcase is exactly what you describe.

We set the Vault max assets to 1,000,000
`tx[sfAssetsMaximum] = BrokerParameters::defaults().vaultDeposit; <=== 1,000,000`

We request principal of 1,000, but the interest pushes the TotalValue above the limit of 1,000,000